server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -921,7 +935,8 @@ public void cleanup(long repositoryStateId, Version repositoryMetaVersion, Actio
                 // delete would also log a confusing INFO message about "stale blobs".
                 final Set<String> blobNamesToIgnore = deletedSnapshots.stream().flatMap(
                         snapshotId -> Stream.of(GLOBAL_METADATA_FORMAT.blobName(snapshotId.getUUID()),
-                                SNAPSHOT_FORMAT.blobName(snapshotId.getUUID()))).collect(Collectors.toSet());
+                            SNAPSHOT_FORMAT.blobName(snapshotId.getUUID()), INDEX_FILE_PREFIX + previousGeneration))
```

Comment:
why are we adding `INDEX_FILE_PREFIX + previousGeneration` here?

Reply:
Ops, I've just seen this https://github.com/elastic/elasticsearch/pull/62152#discussion_r485447600